source/index.ts

```diff
@@ -78,24 +68,66 @@ export async function toPassPackageAudit(
       pass = true;
     } else {
       // Strip ANSI colour encoding.
-      const outputString = output.toString().replace(/\u001b\[.*?m/g, '');
-      let match;
+      const outputString = output
+        .toString()
+        .replace(/\u001b\[.*?m/g, '')
+        .replace(/}\s*{/g, '},{');
+      const correctedOutputString = `[${outputString}]`;
+      const rows = JSON.parse(correctedOutputString);
+
+      const matches = [];
+      for (const row of rows) {
+        if (row !== '') {
+          if (row?.type && row.type === 'auditAdvisory') {
+            // Process yarn audit --json
+            matches.push({
+              pkgName: row.data.advisory.module_name,
+              pkgSeverity: row.data.advisory.severity,
+            });
+          } else if (row?.advisories) {
+            // Process npm audit --json
+            let name = '';
+            let severity = '';
+            for (const key in row.advisories) {
+              const item = row.advisories[key];
+              if (item?.module_name && item?.severity) {
+                name = item.module_name;
+                severity = item.severity;
+              }
+            }
+            if (name && severity) {
+              matches.push({
+                pkgName: name,
```

Comment:
Since these names are potentially exposed to the user via the filter function, can we use the full names e.g:
```suggestion
                packageName: name,
```